src/cfnlint/rules/conditions/Equals.py

```diff
@@ -35,8 +35,8 @@ def _check_equal_values(self, cfn, element, path):
                 elif element_key == 'Ref':
                     valid_refs = cfn.get_valid_refs()
                     if element_value in valid_refs:
-                        if valid_refs[element_value].get('From') == 'Parameters':
-                            if valid_refs[element_value].get('Type') in VALID_PARAMETER_TYPES_LIST:
+                        if valid_refs.get(element_value).get('From') == 'Parameters':
```

Comment:
Good catch.

Reply:
Are we sure this works as expected?

If the key exist, they should behave the same. If it doesn't you get an exception for doing None.get()

Or am I misreading this?

Reply:
yea agreed.  I like `get` but we should probably have a default value to avoid any possible failures.  I actually think the fix you put in solved this issue.  Testing some more

Reply:
It seems like that, by using .get(key) instead of dic[key] I was avoiding the __getitem__ method in the RegexDict class, that's why it "worked" but probably not the way you would expect.

Reply:
Should we implement that method and either:
- throw a NotImplementedError
- actually implement it (catch KeyError, return default)?

Reply:
added this to #1816 
```
def get(self, key, default=None):
        try:
            return self[key]
        except KeyError:
            return default
```